core/src/main/scala/cromwell/core/actor/BatchingDbWriter.scala

```diff
@@ -0,0 +1,50 @@
+package cromwell.core.actor
+
+import cats.data.NonEmptyVector
+import org.slf4j.LoggerFactory
+
+import scala.util.{Failure, Success, Try}
+
+
+/** A collection of state, data, and message types to support batched database writes. */
+object BatchingDbWriter {
+  val logger = LoggerFactory.getLogger("BatchingDbWriteActor")
+
+  /** Data for batched database writes. */
+  sealed trait BatchingDbWriterData {
+    def addData[D](datum: D): BatchingDbWriterData = addData(Vector(datum))
+    def addData[D](data: Iterable[D]): BatchingDbWriterData = {
+      Try(NonEmptyVector.fromVector(data.toVector)) match {
+        case Success(Some(v)) =>
+          val newEvents = this match {
+            case NoData => v
+            case HasData(e) => e.concatNev(v)
+          }
+          HasData(newEvents)
+        case Success(None) => this
+        case Failure(f) =>
+          logger.error("Failed processing batched data. Data will be dropped and not be sent to the database.", f)
```

Comment:
I can imagine someone coming to office hours or the WDL forum saying "my cromwell says its dropped some data. Should I be concerned, worried, or terrified?". It'd be nice to have a better answer than :man_shrugging: